### PR TITLE
Update search docs for v3.5

### PR DESCRIPTION
**Pull request by chriscressman**

### CODE REVIEW

**Comment by chriscressman** [comment]
I was primarily concerned here with updating docs that were invalidated by the changes in 3.5. I'd like to cover the effects of releases and segments on search in more detail, but I'd rather do that as a separate set of work (e.g. when I write docs for releases and segments).

File: docs/source/articles/searching.html.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Workarea also provides query classes which encapsulate the complexity of the var
 
 ## Client & Server(s)
 
-Workarea uses a [Ruby client](http://www.rubydoc.info/gems/elasticsearch-transport/5.0.4/Elasticsearch/Transport/Client) to communicate with an Elasticsearch [cluster](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/_basic_concepts.html#_cluster). Workarea Hosting provisions each cloud environment with a cluster. You must provision your own cluster for local environments.
+Workarea uses a [Ruby client](http://www.rubydoc.info/gems/elasticsearch-transport/5.0.4/Elasticsearch/Transport/Client) to communicate with an Elasticsearch [cluster](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/_basic_concepts.html#_cluster). In Workarea Cloud environments, the Elasticsearch cluster is already provisioned. You must provision your own cluster in other environments.
```
Comment (bencrouse) [inline docs/source/articles/searching.html.md]: Probably doesn't really matter for this, but FYI we use ES 5.6 in production.

File: docs/source/articles/storefront-search-features.html.md [hunk 2]
```diff
@@ -452,7 +450,10 @@ You will see the fields referenced this way within Elasticsearch request bodies
 
 The namespaces (e.g. `active` and `content`) perform two primary functions.
 The `active` namespace is a special case which groups fields representing the activeness of products.
-This field allows for more accurate previewing of releases in the Storefront (inactive products are excluded from results).
+These fields allows for more accurate previewing of releases in the Storefront (inactive products are excluded from results).
+Before Workarea 3.5, these fields keyed off of releases (storing a per-release active value).
+Since Workarea 3.5, they key off of segments.
```
Comment (bencrouse) [inline docs/source/articles/storefront-search-features.html.md]: This won't really be true in the final implementation, I changed it to simplify: https://github.com/workarea-commerce/workarea/blob/make-activeness-per-segment/core/app/models/workarea/search/storefront.rb#L40-L51

File: docs/source/articles/searching.html.md (hunk 1 above)
Comment (chriscressman) [reply docs/source/articles/searching.html.md]: Just to be safe, I updated all links to use 5.6

**Comment by bencrouse** [comment]
Ok @chriscressman , this looks good. But before merging, do you want to wait for the further updates for segmentation (the final PR) or potentially open a second one based on that?

**Comment by chriscressman** [comment]
@bencrouse I'll go ahead and merge this in and keep an eye out for the final changes and open an additional PR if necessary.

**Pull request merged**